ice.c

```diff
@@ -1431,14 +1447,25 @@ static void janus_ice_webrtc_free(janus_ice_handle *handle) {
 		return;
 	}
 	handle->agent_created = 0;
+#ifdef HAVE_CLOSE_ASYNC
+	nice_agent_attach_recv(handle->agent, handle->stream_id, 1, g_main_loop_get_context(handle->mainloop), NULL, NULL);
+#endif
 	if(handle->stream != NULL) {
 		janus_ice_stream_destroy(handle->stream);
 		handle->stream = NULL;
 	}
 	if(handle->agent != NULL) {
+#ifdef HAVE_CLOSE_ASYNC
+		if(G_IS_OBJECT(handle->agent)) {
+			JANUS_LOG(LOG_INFO, "[%"SCNu64"] Closing nice agent %p\n", handle->handle_id, handle->agent);
+			nice_agent_close_async(handle->agent, janus_ice_cb_agent_closed, handle);
+			janus_refcount_increase(&handle->ref);
```

Comment:
Would it make sense to increase the refcount before the `nice_agent_close_async` call (to avoid a race), or are we guaranteed that `janus_ice_cb_agent_closed` will not be called before the end of this block (and thus, there is no race)?

Reply:
The handle is already protected by a mutex.